cyclonedx-bom/src/models/bom.rs

```diff
@@ -30,13 +31,15 @@ use crate::models::external_reference::ExternalReferences;
 use crate::models::metadata::Metadata;
 use crate::models::property::Properties;
 use crate::models::service::{Service, Services};
+use crate::models::vulnerability::Vulnerabilities;
 use crate::validation::{
     FailureReason, Validate, ValidationContext, ValidationError, ValidationPathComponent,
     ValidationResult,
 };
 use crate::xml::{FromXmlDocument, ToXml};
 
-#[derive(Debug, PartialEq, Eq)]
+// todo: derive(Eq)
```

Comment:
Why is this a TODO? Are there any blockers to actually doing this?

Reply:
Yes, there was a detail I was not sure how to handle it appropriately. Spec 1.4 introduced a vulnerability rating with a decimal score. In rust this is a f32 which does not allow to derive `Eq`.

Today I've worked on a solution. I'm not sure though, that my approach is okay. It would be great if you could check it and share your thoughts. Thanks.

Reply:
I like the approach you have taken. Thanks!